src/app/components/search/search.component.ts

```diff
@@ -1,14 +1,51 @@
 import { Component } from '@angular/core';
+import { FormGroup, FormControl } from '@angular/forms';
 
 @Component({
   selector: 'app-search',
   templateUrl: './search.component.html',
   styleUrls: ['./search.component.css'],
 })
 export class SearchComponent {
-  searchText = '';
+  specificShow = false;
+  rangeShow = false;
+
+  form = new FormGroup({
+    searchText: new FormControl(),
+  });
+
+  // Submit Registration Form
+  onSubmit() {
+    alert(JSON.stringify(this.form.value));
+  }
+
+  //Fires on radio input checked change
+  onItemChange(type: string, evt: Event) {
+    console.log(evt);
+    const radio = evt.target as HTMLInputElement;
+    if (evt.target) {
+      if (radio.checked) {
+        this.toggleRadioSelected(true, type);
+      } else {
+        this.toggleRadioSelected(false, type);
+      }
+    }
+  }
```

Comment:
This appears quite complex as it's generic.

You shouldn't need to pass the event in to know if the radio is checked since we store the state in the component. Looking at it further, could you call directly to `toggleRadioSelected` from the template? Something like `toggleRadioSelected('specific')` which can then change the state in the component?